src/components/SinglePage.tsx

```diff
@@ -0,0 +1,56 @@
+import { Component, ErrorInfo, ReactNode } from 'react'
+import { withRouter } from './hoc/createWithrouter'
+import { Link } from 'react-router-dom'
+import { sendRequest } from '../module/sendRequest'
+import IUser from './context/PostsContext'
+import { WithRouterProps } from './hoc/createWithrouter'
+import { POST_PAGE } from './path/path'
+import { URL_POSTS } from '../module/sendRequest'
+type IComponent = Partial<IUser>
+
+interface IState {
+  post: IComponent
+}
+
+class SinglePage extends Component<WithRouterProps, IState> {
+  constructor(props: WithRouterProps) {
+    super(props)
+
+    this.state = {
+      post: {},
+    }
+  }
+  async sendRequestCreateOnePost() {
+    const getItems = await sendRequest({
```

Comment:
Что означает твое название переменной ?) получитьЭлементы ? 

Именование идет с глагола - значит это функция ??

```suggestion
    const post = await sendRequest({
```